### PR TITLE
CLN: indexing comments and cleanups

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -731,11 +731,15 @@ def _getitem_lowerdim(self, tup: Tuple):
...
         for i, key in enumerate(tup):
-            if is_label_like(key) or isinstance(key, tuple):
+            if is_label_like(key):
+                # We don't need to check for tuples here because those are
+                #  caught by the _is_nested_tuple_indexer check above.
                 section = self._getitem_axis(key, axis=i)
 
-                # we have yielded a scalar ?
-                if not is_list_like_indexer(section):
+                if not hasattr(section, "ndim"):
```
Comment (jreback) [inline pandas/core/indexing.py]: hmm, can you then eliminate this?

File: pandas/core/indexing.py [hunk 2]
```diff
@@ -731,11 +731,15 @@ def _getitem_lowerdim(self, tup: Tuple):
             raise IndexingError("Too many indexers. handle elsewhere")
 
         for i, key in enumerate(tup):
-            if is_label_like(key) or isinstance(key, tuple):
+            if is_label_like(key):
```
Comment (jreback) [inline pandas/core/indexing.py]: can you add an assert

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexing.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/indexing.py]: im ambivalent about this, since optimization is a big part of the goal here

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**